sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -477,6 +487,80 @@ public void serializationOnVariousScenarios() {
+                ".channel.AbstractChannel$AnnotatedConnectException: Connection refused: no further information");
+
+            client.close();
+            client = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .directMode()
+                .buildClient();
+            container =
+                client.getDatabase(cosmosAsyncContainer.getDatabase().getId()).getContainer(cosmosAsyncContainer.getId());
+            AsyncDocumentClient asyncDocumentClient = client.asyncClient().getContextClient();
+            GlobalAddressResolver addressResolver = (GlobalAddressResolver) FieldUtils.readField(asyncDocumentClient,
+                "addressResolver", true);
+
+            @SuppressWarnings("rawtypes")
+            Map addressCacheByEndpoint = (Map) FieldUtils.readField(addressResolver,
+                "addressCacheByEndpoint",
+                true);
+            Object endpointCache = addressCacheByEndpoint.values().toArray()[0];
+            GatewayAddressCache addressCache = (GatewayAddressCache) FieldUtils.readField(endpointCache, "addressCache", true);
+
+            HttpClient httpClient = httpClient(true);
+            FieldUtils.writeField(addressCache, "httpClient", httpClient, true);
+            new Thread(() -> {
+                try {
+                    Thread.sleep(5000);
+                    HttpClient httpClient1 = httpClient(false);
+                    FieldUtils.writeField(addressCache, "httpClient", httpClient1, true);
+                } catch (Exception e) {
+                    fail(e.getMessage());
+                }
+            }).start();
+            PartitionKey partitionKey = new PartitionKey(internalObjectNode.get("mypk"));
+            CosmosItemResponse<InternalObjectNode> readResourceResponse =
+                container.readItem(internalObjectNode.getId(), partitionKey, new CosmosItemRequestOptions(),
+                    InternalObjectNode.class);
+
+            //Partial success address resolution client side statistics
+            assertThat(readResourceResponse.getDiagnostics().toString()).contains("addressResolutionStatistics");
+            assertThat(readResourceResponse.getDiagnostics().toString()).contains("\"inflightRequest\":false");
+            assertThat(readResourceResponse.getDiagnostics().toString()).doesNotContain("endTime=\"null\"");
+            assertThat(readResourceResponse.getDiagnostics().toString()).contains("\"errorMessage\":null");
+            assertThat(readResourceResponse.getDiagnostics().toString()).contains("\"errorMessage\":\"io.netty" +
+                ".channel.AbstractChannel$AnnotatedConnectException: Connection refused: no further information");
+        } catch (Exception ex) {
+            fail("This test should not throw exception");
```

Comment:
you don't need to catch exception. and fail. the test will fail regardless of this. if the exception is thrown.